src/main/java/org/cryptomator/ui/preferences/UpdatesPreferencesController.java

```diff
@@ -1,48 +1,91 @@
+import java.util.Locale;
+import java.util.ResourceBundle;
+
 
 @PreferencesScoped
 public class UpdatesPreferencesController implements FxController {
 
 	private static final String DOWNLOADS_URI = "https://cryptomator.org/downloads";
 
 	private final Application application;
+	private final Environment environment;
+	private final ResourceBundle resourceBundle;
 	private final Settings settings;
 	private final UpdateChecker updateChecker;
 	private final ObjectBinding<ContentDisplay> checkForUpdatesButtonState;
 	private final ReadOnlyStringProperty latestVersion;
+	private final ObservableValue<Instant> lastSuccessfulUpdateCheck;
+	private final ObservableValue<String> timeDifferenceMessage;
 	private final String currentVersion;
 	private final BooleanBinding updateAvailable;
+	private final BooleanProperty upToDateLabelVisible = new SimpleBooleanProperty(false);
+	private final ObjectProperty<UpdateChecker.UpdateCheckState> updateCheckState;
+	private final DateTimeFormatter formatter;
 
 	/* FXML */
 	public CheckBox checkForUpdatesCheckbox;
+	public HBox checkFailedHBox;
+	public Label upToDateLabel;
 
 	@Inject
-	UpdatesPreferencesController(Application application, Settings settings, UpdateChecker updateChecker) {
+	UpdatesPreferencesController(Application application, Environment environment, ResourceBundle resourceBundle, Settings settings, UpdateChecker updateChecker) {
 		this.application = application;
+		this.environment = environment;
+		this.resourceBundle = resourceBundle;
 		this.settings = settings;
 		this.updateChecker = updateChecker;
 		this.checkForUpdatesButtonState = Bindings.when(updateChecker.checkingForUpdatesProperty()).then(ContentDisplay.LEFT).otherwise(ContentDisplay.TEXT_ONLY);
 		this.latestVersion = updateChecker.latestVersionProperty();
-		this.updateAvailable = latestVersion.isNotNull();
+		this.lastSuccessfulUpdateCheck = updateChecker.lastSuccessfulUpdateCheckProperty();
+		this.timeDifferenceMessage = lastSuccessfulUpdateCheck.map(this::updateTimeDifferenceMessage);
 		this.currentVersion = updateChecker.getCurrentVersion();
+		this.updateAvailable = updateChecker.updateAvailableProperty();
+		this.updateCheckState = updateChecker.updateCheckStateProperty();
+		this.formatter = DateTimeFormatter.ofLocalizedDateTime(FormatStyle.MEDIUM).withLocale(Locale.getDefault());
 	}
 
 	public void initialize() {
 		checkForUpdatesCheckbox.selectedProperty().bindBidirectional(settings.checkForUpdates);
+
+		BooleanBinding isUpdateSuccessfulAndCurrent = updateCheckState.isEqualTo(UpdateChecker.UpdateCheckState.CHECK_SUCCESSFUL).and(latestVersion.isEqualTo(currentVersion));
+
+		updateCheckState.addListener((_, _, _) -> {
+			if (isUpdateSuccessfulAndCurrent.get()) {
+				upToDateLabelVisible.set(true);
+				PauseTransition delay = new PauseTransition(javafx.util.Duration.seconds(5));
+				delay.setOnFinished(_ -> upToDateLabelVisible.set(false));
+				delay.play();
+			}
+		});
```

Comment:
i just noticed we are listening to the wrong property here. We depend on updateCheckerState _and_ latestVersion. Hence, the local binding should be an instance variable and we should add a listener to this binding, not `updateCheckState`.

This prevents memory leaks are premature garbage collection of our bindings.

Reply:
Also, we do not need to create an extra method imho.